src/codec/builtin/FixSizedTypesCodec.ts

```diff
@@ -43,6 +73,29 @@ export class FixSizedTypesCodec {
         return new Long(low, high);
     }
 
+    static encodeNumberAsLong(buffer: Buffer, offset: number, value: any): void {
+       if (value <= -TWO_PWR_63_DBL) {
+            // MIN_VALUE
+            buffer.writeInt32LE(0, offset);
+            buffer.writeInt32LE(0x80000000|0, offset + BitsUtil.INT_SIZE_IN_BYTES);
+        }
+
+        if (value + 1 >= TWO_PWR_63_DBL) {
+            // MAX_VALUE
+            buffer.writeInt32LE(0xFFFFFFFF|0, offset);
```

Comment:
nit: this and the following `buf.writeInt32LE()` calls could be replaced with a single `buf.copy()` call if we would introduce a const `Buffer` which would hold the `MAX_VALUE`.